### PR TITLE
Adds Sanguine and Osseous Reagent

### PR DESCRIPTION
**Pull request by JimKil3**
## What Does This PR Do
Progresses #20532 by implementing the reagents that would be used to fix internal bleeding and bone breaks. In the interest of atomicity (is that a word?) and because I believe these chemicals are interesting enough to be added on their own, they're being added in a separate PR.

Sanguine reagent, currently, restores blood quickly (up to about five times as fast as iron does), but has a risk of overdosing, causing brute and suffocation damage, so you can't just pump someone full of it and be done. It's made with the following:
- 1u Omnizine
- 2u Synthflesh
- 2u Iron

Osseous reagent, currently, does minor toxin damage. However, with enough in your system (over 30 units), there's a chance that a bone tumor will form because of it, causing lethal damage to your lower body and the organs inside of it. The tumor requires surgery to be removed. While not a medicine, this should provide an interesting & effective alternate to normal poisons which can be countered with just pentetic acid or charcoal. It's made with the following:
- 1u Milk
- 1u Potassium
- 4u Oxygen
- 3u Phenol
- 3u Formaldehyde

Of course, all of these numbers are subject to change if the balance or design teams find something too effective or weak.

## Why It's Good For The Game
In addition to what they'd provide in the context of the cloning rework, both of these chemicals would benefit the game in their own ways. Sanguine reagent would give chemistry more options on what to do with their omnizine; should they make 90 pills of lazarus reagent, or split it between LR and (the new) SR? It also fills a niche of quick, but potentially risky, blood restoration.

Osseous reagent gives traitors or other antagonists a new option for poisoning that isn't countered by charcoal/pentetic acid/calomel. However, the high amount of osseous reagent required to form a bone tumor will need a new strategy than just syringe gunning / shotgun darting someone. On the medical side of things, it's a new, niche problem to deal with that requires surgery - though not a difficult one, as removing the bone tumor is just organ manipulation on the lower body.

## Testing
Loaded in and drank things I probably shouldn't have, checking occasionally with a health analyzer.
Made a skrell donate blood, then gave it some sanguine reagent, periodically checking its blood level to make sure SR works.
Also made a skrell drink a lot of osseous reagent, then removed the tumor that formed as a result, after letting it damage the skrell.

## Changelog
:cl:
add: Added Sanguine Reagent, a potentially-risky way to restore blood quickly
add: Added Osseous Reagent, an insidious poison that requires surgery to treat properly
/:cl:

### CODE REVIEW

File: code/modules/reagents/chemistry/reagents/medicine.dm [hunk 1]
```diff
@@ -809,6 +809,84 @@
...
+
+/datum/reagent/medicine/osseous_reagent/overdose_process(mob/living/M, severity)
+	var/update_flags = STATUS_UPDATE_NONE
+	update_flags |= M.adjustToxLoss(1, FALSE)
+
+	if(prob(5))
+		if(ishuman(M))
+			var/mob/living/carbon/human/H = M
+			if(!H.get_int_organ(/obj/item/organ/internal/bone_tumor))
+				new/obj/item/organ/internal/bone_tumor(H)
```
Comment (Henri215) [inline code/modules/reagents/chemistry/reagents/medicine.dm:886]:
```suggestion
	if(ishuman(M) && prob(5))
		var/mob/living/carbon/human/H = M
		if(!H.get_int_organ(/obj/item/organ/internal/bone_tumor))
			new/obj/item/organ/internal/bone_tumor(H)
```

File: code/modules/reagents/chemistry/reagents/medicine.dm [hunk 2]
```diff
@@ -819,6 +819,83 @@
...
+	name = "Sanguine Reagent"
+	id = "sanguine_reagent"
+	description = "A deeply crimson almost-gel that can mimic blood, regardless of type."
+	color = "#770101"
+	taste_description = "coppery fuel"
+	harmless = FALSE
+	overdose_threshold = 15
+
+/datum/reagent/medicine/sanguine_reagent/on_mob_life(mob/living/M)
+	if(ishuman(M))
```
Comment (farie82) [inline code/modules/reagents/chemistry/reagents/medicine.dm:832]: Can you make this into a guard (early return)?

**Comment by Edan52** [comment]
I think the general consensus is to lower the blood restoration amount. Something stronger than iron but harder to make? Sure, but not five times stronger, might wanna lower it a bit.

**Comment by JimKil3** [comment]
>I think the general consensus is to lower the blood restoration amount. Something stronger than iron but harder to make? Sure, but not five times stronger, might wanna lower it a bit.

This is balanced, in my mind, by the fact that sanguine reagent takes omnizine to make; cutting from the total amount of lazarus reagent you'll be able to make, as well as cutting from how many IBs you'll be able to fix with the initial 30u of omnizine once the cloning rework is in.

If a decrease is still wanted despite that, I'm fine with doing it, just saying my reasoning. Are there any balance concerns with osseous reagent, or is that fine to leave as-is?

**Comment by tipoima** [comment]
Osseous seems just too slow to be useful. 
Tumor takes forever to appear and then takes forever to actually kill the victim. And the toxic damage will cause them to still seek treatment, most likely purging osseous before the tumor forms. 
I just don't see any situation where I'd use it over conventional poisons.

**Comment by CinnamonSnowball** [comment]
> I think the general consensus is to lower the blood restoration amount. Something stronger than iron but harder to make? Sure, but not five times stronger, might wanna lower it a bit.

Not sure where that "consensus" is coming from, but as far as Medbay is concerned, the amount of Sanguine Reagent you get (and thus how much blood you can recover) isn't super noteworthy, only enough for about 3-4 people; if anything, it's used best as an emergency tool for rare types of blood like Vox or Slime Jelly.

Outside of Medbay or if Chemistry wants to make more of it, you *do* have to go through the effort of finding a source of Omnizine, typically requiring interaction with someone to help you out - and in case of antags, additionally either a chemistry setup or some clever thievery - so not like it's more worth it than simply ordering a blood crate (which refills blood even faster)

**Review by S34NW — APPROVED** [review]

**Pull request merged**